src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1507,172 @@ public void addUpdateRemoveConditionalFormat() throws Exception
+        assertNotNull("expected decimal field was not exported", decimalField);
+        assertThat(decimalField.getFormat(), is("0.####"));
+
+        PropertyDescriptor attachmentField = createdFields.stream().filter(a-> a.getName().equals("attachmentField"))
+                .findFirst().orElse(null);
+        assertNotNull("expected attachment field was not exported", attachmentField);
+        assertTrue(attachmentField.getHidden());
+
+        for (PropertyDescriptor intendedField : dgen.getColumns())
+        {
+            PropertyDescriptor exportedField = createdFields.stream().filter(a-> a.getName().equals(intendedField.getName()))
+                    .findFirst().orElse(null);
+            assertNotNull("expected field ["+intendedField.getName()+"] was not exported", exportedField);
+            verifyExpectedFieldProperties(intendedField, exportedField);
+        }
+    }
+
+    @Test
+    public void testFileImportFieldsToNewDomain() throws Exception
+    {
+        String sampleType = "importFieldsTestSampleType";
+
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", sampleType);
+        ConditionalFormatFilter gtFive = new ConditionalFormatFilter("5", Filter.Operator.GT);
+        FieldDefinition.RangeValidator soonValidator = new FieldDefinition.RangeValidator("soonValidator", "between now and 2 days from now", "aaaa",
+                FieldDefinition.RangeType.GTE, "2020-11-04",
+                FieldDefinition.RangeType.LTE, "2020-11-06");
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("name", FieldDefinition.ColumnType.String),
+                        new FieldDefinition("textField", FieldDefinition.ColumnType.String)
+                                .setDescription("is texty").setLabel("text field"),
+                        new FieldDefinition("booleanField", FieldDefinition.ColumnType.Boolean)
+                                .setLabel("Boolean Field")
+                                .setRequired(true)
+                                .setURL("fake/list-def.url"),
+                        new FieldDefinition("attachmentField", FieldDefinition.ColumnType.Attachment)
+                                .setHidden(true),
+                        new FieldDefinition("decimalField", FieldDefinition.ColumnType.Decimal).
+                                setScale(1234).setFormat("0.####"),
+                        new FieldDefinition("intField", FieldDefinition.ColumnType.Integer)
+                                .setMvEnabled(true)
+                                .setConditionalFormats(List.of(new ConditionalFormat(List.of(gtFive), true, true, false))),
+                        new FieldDefinition("dateTimeField", FieldDefinition.ColumnType.DateAndTime)
+                                .setValidators(List.of(soonValidator))));
```

Comment:
Looks like this list of field definitions is a duplicate from the other test case. Could this be factored out to a function or variable to be reused in both places?

Reply:
Yes, but List.of(...) produces an immutable list, and I need to add a 'name' field for sampleTypes, so my refactor produces a mutable list